### DISCUSSION TITLE
Initial Setup

### DISCUSSION THREAD

**Discussion by MauriceB13** [bemasher/rtlamr#165] [Q&A]
Hi All:

I’m new to this RTL-SDR world. I just purchased the R820T. I have it all installed and configured and it appears to be working in AirSpy. My end goal is to try and read my smart meter as indicated above. I have installed GO, GIT, Visual Studio Code with the GitHub extension. I then installed the RTLAMR and RTLAMR-COLLECT. I saw the commands above that are supposedly supposed to show me smart meters in my area. However, when I run that command in the terminal of Visual Studio Core, it doesn’t work. My question is where should I be running commands such as:

~/gocode/bin/rtlamr -msgtype=idm –format=json -msgtype=scm+

Thanks,
Ray

**Comment by MauriceB13** [comment]
Looks like I'm getting closer. I ran the following command:

rtlamr -msgtype=idm --format=json -msgtype=scm+ 

and I received the following error:

main.go:85: Error connecting to spectrum server: dial tcp127.0.0.1:1234 connectex: No connection could be made because the target machine actively refused it.
rcvr.Connect
main.(*Receiver).NewReceiver
main.main
runtime.main
runtime.goexit

What did I miss?

Thanks,
Ray

**Reply by konstk1** [reply]
Do you have rtl_tcp running? It must be running first for rtlamr to connect to.

**Comment by MauriceB13** [comment]
Thanks for the quick reply!

I just figured out that was what I was doing wrong. I have that running now.



 

I then ran rtlamr and this was my result.

 



My goal here is to connect to my electric meter.

 

Thanks,

Ray

**Comment by MauriceB13** [comment]
I got rtl_tcp to run and then start rtl-sdr. I see the connection occur in rtl_tcp.

![rtl_tcp](https://user-images.githubusercontent.com/77756698/106534471-20b38100-64c2-11eb-8405-bd8a92eed4a2.png)

![rtlamr](https://user-images.githubusercontent.com/77756698/106534511-40e34000-64c2-11eb-986d-64da9db6f53e.png)

I can't figure out what I have to do next to see my meter. Is there a parameter I have to enter for my meter ID somewhere?

Thanks,
Ray

**Reply by MauriceB13** [reply]
I managed to use the -samplefile parameter to create a .txt file for the output, but nothing was dumped to the file.

**Reply by bemasher** [reply]
`-samplefile` dumps IQ samples for decoded messages to a file, it's a binary format and will need to be viewed in something like baudline or audacity.

You should probably use `-msgtype=all` while trying to locate a meter, when using more than one decoder, they should be comma-separated e.g. `-msgtype=idm,scm+`

**Reply by MauriceB13** [reply]
Thanks for the info @bemasher ! I tried your suggestions and when using -samplefile, I don't see any file being created. Where would this file be located? I would expect in the RTL-SDR directory. 

I then tried using the following:

rtlamr -msgtype=all -format=json -msgtype=idm,scm+ -samplefile=smartmeter.txt

I saw the file "smartmeter.txt" get created, but it was 0kb and nothing was in it. 

Is my RTL-SDR device not picking up anything?

**Reply by MauriceB13** [reply]
I just ran "rtl_sdr smartmeter" and it seemed to create a file named "smartmeter". This time the file was not empty! I then installed audacity and I was able to import the file. It has a lot of "noise" when I play it. How do I now use audacity to find my meter?

**Reply by bemasher** [reply]
Remember, only use a single instance of each flag. You probably want to try with just this command: `rtlamr -msgtype=all`

**Reply by MauriceB13** [reply]
@bemasher so I went back and ran rtl_tcp and then rtlamr -msgtype=all. I got the following results in the rtl_tcp window.

![rtl_tcp](https://user-images.githubusercontent.com/77756698/108927160-c6639700-760d-11eb-8278-e22a1fba59e2.png)

What does this mean?

Thanks,
Ray

**Reply by bemasher** [reply]
The system you are running rtlamr on isn't powerful enough to process all of the samples being produced by your radio. -msgtype=all is designed for discovery, once you know what message types your meters are transmitting, you should set it to only decode those types. It is likely that the R900 decoder is eating up all of your processing power, if you don't need it, set -msgtype to only the types you need.
